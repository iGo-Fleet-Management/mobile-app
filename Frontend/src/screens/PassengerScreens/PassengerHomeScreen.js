import React, { useState } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

// Components
import Header from '../../components/common/Header';
import UserIcon from '../../components/common/UserIcon';
import TravelModeSelector from '../../components/home/TravelModeSelector';
import StatusSwitch from '../../components/home/StatusSwitch';
import AlertBox from '../../components/home/AlertBox';
import MapContainer from '../../components/home/MapContainer';
import BottomUserBar from '../../components/home/BottomUserBar';

// Assets
import MapImage from '../../../assets/images/google-map-example-blog.png';

const PassengerHomeScreen = ({ navigation }) => {
  const [travelMode, setTravelMode] = useState('roundTrip');
  const [isLiberado, setIsLiberado] = useState(false);

  const handleUserIconPress = () => {
    navigation.navigate('Profile');
  };

  return (
    <SafeAreaView style={styles.container} edges={['top']} testID="passenger-home-screen">
      <View style={styles.container} testID="passenger-home-container">
        {/* Header Section */}
        <View style={styles.headerContainer} testID="header-container">
          <Header 
            title="iGO" 
            testID="app-header"
          />
          <View style={styles.userIconWrapper}>
            <UserIcon 
              onPress={handleUserIconPress} 
              userName="John" 
              testID="user-icon"
            />
          </View>
        </View>

        {/* Date and Travel Mode Section */}
        <View style={styles.dateCard} testID="date-card">
          <Text style={styles.dayOfWeek} testID="day-text">Segunda-Feira</Text>
          <Text style={styles.date} testID="date-text">27 de Novembro de 2023</Text>
          
          <TravelModeSelector 
            selectedMode={travelMode}
            onSelectMode={setTravelMode}
            testID="travel-mode-selector"
          />
        </View>

        {/* Status Switch */}
        <StatusSwitch 
          value={isLiberado}
          onValueChange={setIsLiberado}
          onHelpPress={() => navigation.navigate('Ajuda')}
          testID="status-switch"
        />

        {/* Alert Box */}
        <AlertBox 
          message="Seu motorista jÃ¡ iniciou o trajeto. Fique atento!"
          onEditPress={() => {}}
          testID="alert-box"
        />

        {/* Map Container */}
        <MapContainer 
          source={MapImage}
          testID="map-container"
        />

        {/* Bottom User Bar */}
        <BottomUserBar 
          userName="John Doe"
          testID="bottom-user-bar"
        />
      </View>
    </SafeAreaView>
  );
};
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f5f5f5',
  },
  headerContainer: {
    position: 'relative',
  },
  userIconWrapper: {
    position: 'absolute',
    right: 15,
    top: 0,
    height: 60,
    justifyContent: 'center',
  },
  dateCard: {
    backgroundColor: '#ffffff',
    borderRadius: 8,
    padding: 15,
    margin: 10,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 2,
  },
  dayOfWeek: {
    fontSize: 18,
    fontWeight: 'bold',
    textAlign: 'center',
  },
  date: {
    fontSize: 14,
    color: '#757575',
    textAlign: 'center',
    marginBottom: 15,
  },
});

export default PassengerHomeScreen;