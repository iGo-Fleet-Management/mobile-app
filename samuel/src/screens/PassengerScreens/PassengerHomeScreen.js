import React, { useState } from 'react';
import { View, Text, Image, StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import UserIcon from '../../components/common/UserIcon';

// Components
import Header from '../../components/common/Header';
import TravelModeSelector from '../../components/home/TravelModeSelector';
import StatusSwitch from '../../components/home/StatusSwitch';
import AlertBox from '../../components/home/AlertBox';
import MapContainer from '../../components/home/MapContainer';
import BottomUserBar from '../../components/home/BottomUserBar';

// Assets
import MapImage from '../../../assets/images/google-map-example-blog.png';


export default function HomeScreen({ navigation }) {
  const [travelMode, setTravelMode] = useState('roundTrip');
  const [isLiberado, setIsLiberado] = useState(false);

  const handleUserIconPress = () => {
    navigation.navigate('Profile');
  };

  return (
    <SafeAreaView style={styles.container} edges={['top']}>
      <View style={styles.container}>
        <View style={styles.headerContainer}>
          <Header 
            title="iGO" 
            
          />
          <View style={styles.userIconWrapper}>
            <UserIcon onPress={handleUserIconPress} userName="John" />
          </View>
        </View>

        <View style={styles.dateCard}>
          <Text style={styles.dayOfWeek}>Segunda-Feira</Text>
          <Text style={styles.date}>27 de Novembro de 2023</Text>
          
          <TravelModeSelector 
            selectedMode={travelMode}
            onSelectMode={setTravelMode}
          />
        </View>

        <StatusSwitch 
          value={isLiberado}
          onValueChange={setIsLiberado}
          onHelpPress={() => navigation.navigate('Ajuda')}
        />

        <AlertBox 
          message="Seu motorista jÃ¡ iniciou o trajeto. Fique atento!"
          onEditPress={() => {/* Handle edit */}}
        />

        <MapContainer 
          source={MapImage}
        />

        <BottomUserBar 
          userName="John Doe"
        />
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f5f5f5',
  },
  headerContainer: {
    position: 'relative',
  },
  userIconWrapper: {
    position: 'absolute',
    right: 15,
    top: 0,
    height: 60,
    justifyContent: 'center',
  },
  dateCard: {
    backgroundColor: '#ffffff',
    borderRadius: 8,
    padding: 15,
    margin: 10,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 2,
  },
  dayOfWeek: {
    fontSize: 18,
    fontWeight: 'bold',
    textAlign: 'center',
  },
  date: {
    fontSize: 14,
    color: '#757575',
    textAlign: 'center',
    marginBottom: 15,
  },
});