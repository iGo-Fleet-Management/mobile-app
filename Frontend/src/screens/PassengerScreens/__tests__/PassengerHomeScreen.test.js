import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import PassengerHomeScreen from '../PassengerHomeScreen'; 

// Mock dos componentes
jest.mock('../../../components/common/Header', () => {
  const { Text, View } = require('react-native');
  return ({ title, testID }) => (
    <View testID={testID}>
      <Text>{title}</Text>
    </View>
  );
});

jest.mock('../../../components/common/UserIcon', () => {
  const { TouchableOpacity, Text } = require('react-native');
  return ({ onPress, userName, testID }) => (
    <TouchableOpacity testID={testID} onPress={onPress}>
      <Text>{userName}</Text>
    </TouchableOpacity>
  );
});

jest.mock('../../../components/home/TravelModeSelector', () => {
  const { View, Text } = require('react-native');
  return ({ testID }) => (
    <View testID={testID}>
      <Text>TravelModeSelector</Text>
    </View>
  );
});

jest.mock('../../../components/home/StatusSwitch', () => {
  const { View, Text, TouchableOpacity } = require('react-native');
  return ({ value, onHelpPress, testID }) => (
    <View testID={testID}>
      <Text>{value ? 'Liberado' : 'Não Liberado'}</Text>
      <TouchableOpacity testID={`${testID}-help-button`} onPress={onHelpPress}>
        <Text>Ajuda</Text>
      </TouchableOpacity>
    </View>
  );
});

jest.mock('../../../components/home/AlertBox', () => {
  const { View, Text } = require('react-native');
  return ({ message, testID }) => (
    <View testID={testID}>
      <Text>{message}</Text>
    </View>
  );
});

jest.mock('../../../components/home/MapContainer', () => {
  const { View } = require('react-native');
  return ({ testID }) => <View testID={testID} />;
});

jest.mock('../../../components/home/BottomUserBar', () => {
  const { View, Text } = require('react-native');
  return ({ userName, testID }) => (
    <View testID={testID}>
      <Text>{userName}</Text>
    </View>
  );
});

jest.mock('react-native-safe-area-context', () => ({
  SafeAreaView: ({ children }) => children,
  useSafeAreaInsets: () => ({ top: 0, right: 0, bottom: 0, left: 0 })
}));

describe('PassengerHomeScreen', () => {
  const mockNavigation = {
    navigate: jest.fn(),
    goBack: jest.fn()
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });


  it('deve mostrar a data e dia da semana', () => {
    const { getByTestId } = render(
      <PassengerHomeScreen navigation={mockNavigation} />
    );
    
    expect(getByTestId('day-text').props.children).toBe('Segunda-Feira');
    expect(getByTestId('date-text').props.children).toBe('27 de Novembro de 2023');
  });

  it('deve navegar para o perfil ao pressionar o ícone do usuário', () => {
    const { getByTestId } = render(
      <PassengerHomeScreen navigation={mockNavigation} />
    );
    
    fireEvent.press(getByTestId('user-icon'));
    expect(mockNavigation.navigate).toHaveBeenCalledWith('Profile');
  });

  it('deve navegar para ajuda ao pressionar o botão de ajuda', () => {
    const { getByTestId } = render(
      <PassengerHomeScreen navigation={mockNavigation} />
    );
    
    fireEvent.press(getByTestId('status-switch-help-button'));
    expect(mockNavigation.navigate).toHaveBeenCalledWith('Ajuda');
  });

  it('deve mostrar a mensagem de alerta', () => {
    const { getByTestId } = render(
      <PassengerHomeScreen navigation={mockNavigation} />
    );
    
    expect(getByTestId('alert-box').children[0].props.children)
      .toBe('Seu motorista já iniciou o trajeto. Fique atento!');
  });

  it('deve renderizar todos os componentes principais', () => {
    const { getByTestId } = render(
      <PassengerHomeScreen navigation={mockNavigation} />
    );
    
    expect(getByTestId('travel-mode-selector')).toBeTruthy();
    expect(getByTestId('status-switch')).toBeTruthy();
    expect(getByTestId('map-container')).toBeTruthy();
    expect(getByTestId('bottom-user-bar')).toBeTruthy();
  });

  it('deve corresponder ao snapshot', () => {
    const { toJSON } = render(
      <PassengerHomeScreen navigation={mockNavigation} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});