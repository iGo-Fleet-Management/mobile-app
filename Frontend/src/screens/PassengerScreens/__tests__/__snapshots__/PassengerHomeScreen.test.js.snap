// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PassengerHomeScreen deve corresponder ao snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
    }
  }
  testID="passenger-home-container"
>
  <View
    style={
      {
        "position": "relative",
      }
    }
    testID="header-container"
  >
    <View
      testID="app-header"
    >
      <Text>
        iGO
      </Text>
    </View>
    <View
      style={
        {
          "height": 60,
          "justifyContent": "center",
          "position": "absolute",
          "right": 15,
          "top": 0,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="user-icon"
      >
        <Text>
          John
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 8,
        "elevation": 2,
        "margin": 10,
        "padding": 15,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 4,
      }
    }
    testID="date-card"
  >
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      testID="day-text"
    >
      Segunda-Feira
    </Text>
    <Text
      style={
        {
          "color": "#757575",
          "fontSize": 14,
          "marginBottom": 15,
          "textAlign": "center",
        }
      }
      testID="date-text"
    >
      27 de Novembro de 2023
    </Text>
    <View
      testID="travel-mode-selector"
    >
      <Text>
        TravelModeSelector
      </Text>
    </View>
  </View>
  <View
    testID="status-switch"
  >
    <Text>
      Não Liberado
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="status-switch-help-button"
    >
      <Text>
        Ajuda
      </Text>
    </View>
  </View>
  <View
    testID="alert-box"
  >
    <Text>
      Seu motorista já iniciou o trajeto. Fique atento!
    </Text>
  </View>
  <View
    testID="map-container"
  />
  <View
    testID="bottom-user-bar"
  >
    <Text>
      John Doe
    </Text>
  </View>
</View>
`;
